extensions/wikia/SpecialHealthcheck/HealthController.class.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+/**
+ * Extra health checks controller.
+ *
+ * Currently supports requests for getting the status of database clusters
+ */
+class HealthController extends WikiaController {
+
+	private $current = '';
+
+	private $clusters = [ ];
+	private $errors = [ ];
+	private $messages = [ ];
+	private $status = [ ];
+
+	/**
+	 * Get status of database clusters
+	 *
+	 * Returns:
+	 * - clusters: associative array of cluster statuses
+	 *   cluster name is the key and one of "ok", "warning", "critical" as a value
+	 * - messages: list of notices
+	 * - errors: list of errors
+	 */
+	public function databases() {
+		$cluster = $this->getVal( 'cluster' );
+		$clusters = $this->getAllClusters();
+		if ( $cluster ) {
+			$clusters = array_intersect( [ $cluster ], $clusters );
+		}
+
+		$this->clusters = $clusters;
+
+		$this->testClusters();
+
+		$this->setVal( 'clusters', $this->status );
+		if ( $this->errors ) {
+			$this->setVal( 'errors', $this->errors );
+		}
+		if ( $this->messages ) {
+			$this->setVal( 'messages', $this->messages );
+		}
+
+		$this->setVal('readWrite',[
+			'status' => !wfReadOnly(),
+			'reason' => wfReadOnlyReason(),
+		]);
+	}
+
+	/**
+	 * Execute checks for all requested clusters
+	 */
+	protected function testClusters() {
+		foreach ( $this->clusters as $clusterName ) {
+			$this->current = "{$clusterName}: ";
+
+			$fullHealth = false;
+			$operational = false;
+			try {
+				$databaseName = $this->getClusterDatabase( $clusterName );
+				$loadBalancer = wfGetLB( $databaseName );
+				$serverCount = $loadBalancer->getServerCount();
+
+				$roles = [
+					'master' => [ ],
+					'slave' => [ ],
+				];
+				for ( $i = 0; $i < $serverCount; $i++ ) {
+					$serverName = $loadBalancer->getServerName( $i );
+					$this->current = "{$clusterName}: {$serverName}: ";
+
+					$role = $i == 0 ? 'master' : 'slave';
```

Comment:
I prefer `===` operator in PHP. Do you think using one here will be over-engineering or just following a good habit and leaving a better secured code just-in-case?